TrackModel/src/trackmodel/model/Switch.java

```diff
@@ -1,96 +1,71 @@
+   * @param previous The previous block in the track
+   * @param next1 The first of two possible next blocks in the track
+   * @param next2 The second of two possible next blocks in the track
+   * @param leftDoors If the track is on the left of the track
+   * @param rightDoors If the station is on the right of the track
+   */
+  public Switch(String line, String section, String number, float length,
+                float grade, int speedLimit, String infrastructure, float elevation,
+                float cumElevation, boolean biDirectional, int previous, int next1,
+                int next2, boolean leftDoors, boolean rightDoors) {
+
+    super(line, section, number, length, grade, speedLimit, infrastructure, elevation,
+        cumElevation, biDirectional, previous, next1, leftDoors, rightDoors);
+    this.next2 = next2;
   }
 
-  public void setDestination2(Block dest2) {
-    this.destination2 = dest2;
+  /**
+   * This method will set the other possible next block.
+   * @param newNextBlock A Block will be passed in for the other possible next block.
+   */
+  public void setNextBlockOne(int newNextBlock) {
+    next2 = newNextBlock;
   }
 
   /**
-   * This method will allow the user to set a State.
-   * @param newState The boolean value of a state for the switch.
+   * This method will return the next block in the track.
+   * @return A integer will be returned
    */
-  public void setState(boolean newState) {
-    this.state = newState;
-
-    if (!state) {
-      if (mainB.getNextBlock() == destination1) {
-        mainB.setNextBlock(destination1);
-      } else if (mainB.getPreviousBlock() == destination2) {
-        mainB.setPreviousBlock(destination1);
-      }
-
-      if (destination1.getNextBlock() == null) {
-        destination1.setNextBlock(mainB);
-      } else if (destination1.getPreviousBlock() == null) {
-        destination1.setPreviousBlock(mainB);
-      }
-    } else if (state) {
-      if (mainB.getNextBlock() == destination1) {
-        mainB.setNextBlock(destination2);
-      } else if (mainB.getPreviousBlock() == destination1) {
-        mainB.setPreviousBlock(destination2);
-      }
-
-      if (destination1.getNextBlock() == mainB) {
-        destination1.setNextBlock(null);
-      } else if (destination1.getPreviousBlock() == mainB) {
-        destination1.setPreviousBlock(null);
-      }
+  public int getNextBlock() {
+    return next2;
+  }
 
-      if (destination2.getNextBlock() == mainB) {
-        destination2.setNextBlock(mainB);
-      } else if (destination2.getPreviousBlock() == mainB) {
-        destination2.setPreviousBlock(mainB);
-      }
+  /**
+  * This method will update the status of the switch.
+  * The status will be set to a block
+  * @param newStatus This is the block number that the switch is to be set too
+  */
+  public void setStatus(int newStatus) {
+    if (newStatus == getNextBlock() || newStatus == next2) {
+      status = newStatus;

```

Comment:
getNextBlock() should return a Block object. This should use getNextBlock().getId();

Reply:
I only want to pass an integer for the same reason as getNextBlock().